src/processor/queue-consumer.service.ts

```diff
@@ -85,21 +85,25 @@ export class QueueConsumerService extends WorkerHost implements OnApplicationBoo
 
     try {
       const totalCapacityUsed = await this.graphSdkService.updateUserGraph(job.data.dsnpId, job.data.providerId, job.data.processTransitiveUpdates);
-      await this.setEpochCapacity(totalCapacityUsed);
+      if (Object.keys(totalCapacityUsed).length > 0) {
+        await this.setEpochCapacity(totalCapacityUsed);
+      }
       this.logger.verbose(`Successfully completed job ${job.id}`);
     } catch (e) {
       this.logger.error(`Job ${job.id} failed (attempts=${job.attemptsMade})`);
-      if(e instanceof CapacityLowError){
+      if((e instanceof CapacityLowError || e instanceof UnknownError || e instanceof StaleHashError) && (job.attemptsMade < (job.opts.attempts ?? 3))) {
```

Comment:
nit: refactor magic number `3` to a constant

Reply:
since we only queue once with dead letter, assuming we would not need more than 1 attempt in general to re-queue and any attempt more than that is a failure worth looking at